src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -215,6 +219,7 @@ private unsafe Socket(SafeSocketHandle handle, bool loadPropertiesFromHandle)
                                 }
 
                                 _isConnected = true;
+                                UpdateLocalEndPointOnConnect();
```

Comment:
This is a constructor. I can't see any prior code assigning `_localEndpoint` before this call. In that case we don't need this.

Reply:
Fixed